sympy/printing/fcode.py

```diff
@@ -178,6 +172,32 @@ def _print_Piecewise(self, expr):
                                       "inline operators is not supported in "
                                       "standards earlier than Fortran95.")
 
+    def _print_Assignment(self, expr):
+        from sympy.functions import sign
+        if isinstance(expr.rhs, sign) and self._settings["standard"] < 95:
+            from sympy import Piecewise, Abs
+            arg, = expr.rhs.args
+            if arg.is_integer:
+                args = (0, Eq(arg, 0)), (isign(1, arg), True)
+            elif arg.is_complex:
+                args = (cmplx(0, 0), Eq(Abs(arg), 0)), (arg/Abs(arg), True)
+            else:
+                args = (0, Eq(arg, 0)), (dsign(1, arg), True)
+            return self._print(Piecewise(*[(Assignment(expr.lhs, e), c) for e, c in args]))
+        return super(FCodePrinter, self)._print_Assignment(expr)
+
+    def _print_sign(self, expr):
+        if self._settings["standard"] < 95:
```

Comment:
@bjodah 
I think you should add for version<95 as suggested by @Franjov by adding
```
if expr.has(Assignment):
#code here
```
as I did because now there won't be any tension about assign because as you have overrided now; and it's a good idea.